pkg/registry/deployment/strategy.go

```diff
@@ -99,6 +100,26 @@ func (deploymentStatusStrategy) ValidateUpdate(ctx api.Context, obj, old runtime
 	return validation.ValidateDeploymentUpdate(obj.(*extensions.Deployment), old.(*extensions.Deployment))
 }
 
+type deploymentRollbackStrategy struct {
+	deploymentStrategy
+}
+
+var RollbackStrategy = deploymentRollbackStrategy{Strategy}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update of status
+func (deploymentRollbackStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newDeployment := obj.(*extensions.Deployment)
+	oldDeployment := old.(*extensions.Deployment)
+	newRollback := newDeployment.Spec.Rollback
```

Comment:
Rather than posting the whole resource with just the rollback field changed, I'd prefer to add a new resource type with just the rollback config, and copy the rollback config into DeploymentSpec.rollback.